prometheus/reporter_test.go

```diff
@@ -358,6 +358,29 @@ func TestOnRegisterError(t *testing.T) {
 	assert.Equal(t, 2, len(captured))
 }
 
+func TestAlreadyRegisteredCounter(t *testing.T) {
+	var captured []error
+
+	registry := prom.NewRegistry()
+	r := NewReporter(Options{
+		Registerer: registry,
+		OnRegisterError: func(err error) {
+			captured = append(captured, err)
+		},
+	})
+
+	r.AllocateCounter("foo", map[string]string{"foo": "bar"})
+	r.AllocateCounter("foo", map[string]string{"foo": "baz"}) // ok; same label names
+	r.AllocateCounter("foo", map[string]string{"bar": "qux"}) // errors; different label names
+	r.AllocateCounter("foo", map[string]string{
+		"bar":  "qux",
+		"zork": "derp",
+	}) // errors; different label names
+	r.AllocateCounter("foo", nil) // errors; different label names
```

Comment:
it's not obvious from `different label names ` why this is an error.

can we assert on an error string directly here?

Reply:
sure! added some docs as well.